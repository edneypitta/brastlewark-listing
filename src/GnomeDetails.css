.gnome-details ul {
  margin: 20px auto;
  padding: 0;
}

.gnome-details ul li {
  list-style: none;
  text-decoration: none;
  margin-bottom: 10px;
}
.gnome-details ul li span {
  display: inline-block; 
  width: 140px;
}

.gnome-details .header {
  display: flex;
  align-items: center;
  font-size: calc(10px + 2vmin);
}

.gnome-details .header :nth-child(1) {
  margin-right: 20px;
}

.gnome-details .info {
  color:#333;  
  font-size: calc(1px + 2vmin);
}