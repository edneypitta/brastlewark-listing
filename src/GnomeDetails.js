import React from 'react'
import PropTypes from 'prop-types'
import './GnomeDetails.css'
import Avatar from './shared/Avatar'

const GnomeDetails = ({
  gnome: {
    thumbnail,
    name,
    age,
    weight,
    height,
    hair_color,
    professions = [],
    friends = [] } }) =>
  <div className="gnome-details">
    <div className="header">
      <div>
        <Avatar
          src={thumbnail}
          alt="Gnome Thumbnail" />
      </div>
      <div>{name}</div>
    </div>
    <div className="info">
      <ul>
        <li><span>Age: </span>{age}</li>
        <li><span>Weight: </span>{weight}</li>
        <li><span>Height: </span>{height}</li>
        <li><span>Hair Color: </span>{hair_color}</li>
        <li><span>Professions: </span>{professions.join(', ')}</li>
        <li><span>Friends: </span>{friends.join(', ')}</li>
      </ul>
    </div>
  </div>

GnomeDetails.propTypes = {
  gnome: PropTypes.object.isRequired
}

export default GnomeDetails