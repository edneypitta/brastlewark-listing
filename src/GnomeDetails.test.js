import React from 'react'
import { shallow } from 'enzyme'
import GnomeDetails from './GnomeDetails'

describe('GnomeDetails', () => {
  it('renders without crashing', () => {
    const component = shallow(
      <GnomeDetails
        gnome={{ name: 'gnome 1' }} />)
        
    expect(component).toMatchSnapshot()
  })
})
