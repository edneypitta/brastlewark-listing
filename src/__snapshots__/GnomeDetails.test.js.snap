// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GnomeDetails renders without crashing 1`] = `
<div
  className="gnome-details"
>
  <div
    className="header"
  >
    <div>
      <Avatar
        alt="Gnome Thumbnail"
      />
    </div>
    <div>
      gnome 1
    </div>
  </div>
  <div
    className="info"
  >
    <ul>
      <li>
        <span>
          Age: 
        </span>
      </li>
      <li>
        <span>
          Weight: 
        </span>
      </li>
      <li>
        <span>
          Height: 
        </span>
      </li>
      <li>
        <span>
          Hair Color: 
        </span>
      </li>
      <li>
        <span>
          Professions: 
        </span>
      </li>
      <li>
        <span>
          Friends: 
        </span>
      </li>
    </ul>
  </div>
</div>
`;
